shared/bash_remediation_functions/lineinfile.sh

```diff
@@ -0,0 +1,106 @@
+# Functions which minic Ansible's lineinfile module.
+# dependencies: die.sh, variables.sh
+
+function lineinfile() {
+    local path="$1"
+    local regex="$2"
+    local new_line="$3"
+    local create="$(normalize_boolean "$4" "false")"
+    local state="$(normalize_state "$5" "present")"
+    local insert_after="$6"
+    local insert_before="$7"
+    local insensitive="$(normalize_boolean "$8" "true")"
+
+    [ ! -e "$path" ] && [ "$create" != "true" ] && die "Path '$path' wasn't found on this system and not creating. Refusing to continue."
+    [ ! -e "$path" ] && [ "$create" == "true" ] && touch "$path"
+
+    if [ "$state" == "absent" ]; then
```

Comment:
The complexity of this function is too big. I guess that we need `lineinfile_present`, `lineinfile_absent` etc. We will need a couple of small functions that are unit-testable e.g. using [BATS](https://github.com/bats-core/bats-core).

Reply:
@matejak And you mean to have `lineinfile` macro use `lineinfile_present`and `lineinfile_absent`?
This makes sense to me.

Reply:
I would be OK to have the `linenfile` function calling other functions - my main objection was against complex functions.
What you propose is one step further - IIUC the macro would use and expand to only a subset of the bash code. That's of course even better.

Reply:
IMO, I wouldn't turn this into an Ansible macro. I do like splitting it out into functions. I could make the `present` cases into separate functions as well if you'd like. 

Most configuration files will use both (`_absent` and `_present`) anyways: you'll usually want to remove conflicting values before adding the desired value.

Reply:
Looks good to me.